pkg/skaffold/runner/runner.go

```diff
@@ -135,32 +135,37 @@ func (r *SkaffoldRunner) Run() error {
 	return nil
 }
 
-func (r *SkaffoldRunner) dev() error {
-	// First rebuild everything.
-	bRes, _, err := r.run(r.config.Build.Artifacts)
-	if err != nil {
-		// In dev mode, we only warn on pipeline errors
-		logrus.Warnf("run: %s", err)
-	}
+func (r *SkaffoldRunner) dev(artifacts []*config.Artifact) error {
+	ctx := context.TODO()
+
+	go r.Watch(ctx, artifacts, r.fileChanges)
+
 	for {
-		if bRes != nil {
-			for _, b := range bRes.Builds {
-				tag := b.Tag
-				go kubernetes.StreamLogsRetry(r.opts.Output, r.kubeclient.CoreV1(), tag, 5)
-			}
-		}
-		evt, err := r.Watch(r.config.Build.Artifacts, r.watchReady, r.cancel)
-		if err != nil {
-			return errors.Wrap(err, "running watch")
-		}
-		if evt.EventType == watch.WatchStop {
+		fmt.Println("WAIT") // ??
+		// Wait for next change
+		evt := <-r.fileChanges
+		fmt.Println(evt.EventType) // ??
+		switch {
+		case evt.EventType == watch.WatchError:
```

Comment:
I'm not sure the watch.Event is the right design.

WatchStart and WatchStop are "meta" events without an artifact
WatchError and WatchChange correspond to a single artifact, although they contain multiple "changed artifacts"

Should we get as granular to return the actual notify events? (create, delete, change) We might need this if we're going to start working on "smarter" context uploads and streaming.

I think I did this initially to cut down on how many channels we needed to pass around (one for top level watch events, and one for individual changes)

Do we expose the file changes channel separately from the watch ready/not ready/canceled channel?